app-frontend/src/app/components/map/drawToolbar/drawToolbar.module.js

```diff
@@ -1,7 +1,298 @@
+            this.polygonLayers.push(drawLayer._layers[layer]);
+        });
+        /* eslint-enable no-underscore-dangle */
+
+        this.drawHandler = new L.Draw.Polygon(mapWrapper.map, {
+            allowIntersection: false,
+            shapeOptions: {
+                weight: 2,
+                fillOpacity: 0.2
+            }
+        });
+
+        this.editHandler = new L.EditToolbar.Edit(mapWrapper.map, {
+            featureGroup: drawLayer,
+            // TODO figure out why this option doesn't work
+            allowIntersection: false
+        });
+
+        this.deleteHandler = new L.EditToolbar.Delete(mapWrapper.map, {
+            featureGroup: drawLayer
+        });
+    }
+
+    $onDestroy() {
+        // clear map listeners
+        this.getMap().then((mapWrapper) => {
+            this.listeners.forEach((listener) => {
+                mapWrapper.off(listener);
+            });
+            mapWrapper.deleteLayers('draw');
+        });
+    }
+
+    $onChanges(changes) {
+        if (changes.options && changes.options.currentValue) {
+            this.options = changes.options.currentValue;
+        }
+
+        if (changes.geom && changes.geom.currentValue) {
+            let geom = changes.geom.currentValue.geom;
+            if (!geom || !geom.coordinates) {
+                this.$log.error('Invalid geometry used in draw toolbar');
+            } else {
+                // split multipolygon into polygons
+                let polygons = geom.coordinates.map((coords) => {
+                    return {
+                        type: 'Polygon',
+                        coordinates: coords
+                    };
+                });
+                if (polygons.length) {
+                    this.getMap().then((mapWrapper) => {
+                        this.setDrawLayer(mapWrapper, polygons);
+                    });
+                }
+            }
+        }
+    }
+
+    updatePolygons() {
+        // Coordinates aren't actually wound counter-clockwise,
+        // but our backend doesn't care
+        let polygons = this.polygonLayers.map((layer) => {
+            let feature = layer.toGeoJSON();
+            let coordinates = feature.geometry.coordinates;
+            return coordinates;
+        });
+
+        this.$parent.$ctrl.updateProjectAOIs(polygons);
+    }
+
+    getMultiPolygonFromLayers() {
+        if (this.polygonLayers.length) {
+            return {
+                geom: {
+                    type: 'MultiPolygon',
+                    coordinates: this.polygonLayers.map((layer) => {
+                        let feature = layer.toGeoJSON();
+                        return feature.geometry.coordinates;
+                    })
+                },
+                srid: 3857
+            };
+        }
+        return null;
+    }
+
+    onDoneClicked() {
+        this.onSave({polygons: this.getMultiPolygonFromLayers()});
+    }
+
+    onCancelClicked() {
```

Comment:
I wonder if not creating the annotation at all when the user hits cancel would be more intuitive than creating an unlabeled annotation.